servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala

```diff
@@ -133,7 +133,7 @@ class Http4sServlet(service: HttpService,
   private def toRequest(req: HttpServletRequest): ParseResult[Request] =
     for {
       method <- Method.fromString(req.getMethod)
-      uri <- Uri.requestTarget(s"${req.getRequestURI}?${req.getQueryString}")
+      uri <- Uri.requestTarget(Option(req.getQueryString).map { q => s"${req.getRequestURI}?$q" }.getOrElse(req.getRequestURI))
```

Comment:
This gets called on every request.  I wonder if a `fold` or a pattern match is faster.  I'll try a quick jmh experiment.